### PR TITLE
fix: components relationship update failed

### PR DESCRIPTION
**Pull request by alexcodelf**
**Problem:**

Updating resource components relationships can lead to the violation of foreign key constraints. This issue arises during the update of resource components (see https://github.com/seal-io/walrus/blob/main/pkg/apis/resourcerevision/extension.go#L200). In the update process, old resources get deleted. However, if a relationship was created using a resource prior to its deletion, it can result in a query failure.

**Solution:**

Clear removed resource components before create components relationships.

**Related Issue:**
#2049

### CODE REVIEW

File: pkg/dao/resource_component.go [hunk 1]
```diff
@@ -116,3 +118,36 @@ func ResourceComponentGetUniqueKey(r *model.ResourceComponent) string {
 	// Align to schema definition.
 	return strs.Join("-", string(r.ConnectorID), r.Shape, r.Mode, r.Type, r.Name)
 }
+
+// clearComponents clears the components that not exist in the database.
+func clearComponents(
+	ctx context.Context,
+	mc model.ClientSet,
+	components model.ResourceComponents,
+) (model.ResourceComponents, error) {
```
Comment (thxCode) [inline pkg/dao/resource_component.go:127]: i didn't see any delete/remove action in this function, can we clarify the purpose of this function?

**Comment by thxCode** [comment]
Can we try to get a clean component slice from outside?

File: pkg/dao/resource_component.go (hunk 1 above)
Comment (alexcodelf) [reply pkg/dao/resource_component.go]: updated, we need refactor the resource components update logic in 0.6.0.

**Review by thxCode — APPROVED** [review]
LGTM

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**